providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/grpc/GrpcStreamConnector.java

```diff
@@ -123,8 +125,29 @@ static void observeEventStream(final BlockingQueue<StreamPayload> writeTo,
         log.info("Initializing sync stream observer");
 
         while (!shutdown.get()) {
+            Exception metadataException = null;
             log.debug("Initializing sync stream request");
-            serviceStub.syncFlags(request, new GrpcStreamHandler(streamReceiver));
+            final SyncFlagsRequest.Builder syncRequest = SyncFlagsRequest.newBuilder();
+            final GetMetadataRequest.Builder metadataRequest = GetMetadataRequest.newBuilder();
+            Map<String, Object> metadata = Collections.emptyMap();
+
+            if (selector != null) {
+                syncRequest.setSelector(selector);
+            }
+
+            serviceStub.withDeadlineAfter(deadline, TimeUnit.MILLISECONDS).syncFlags(syncRequest.build(),
```

Comment:
@guidobrei @aepfli I found we weren't using the passed deadline on either of the gRPC calls in this class, so pass that along as well and added related assertions.